src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/vedtaksperiode/VedtaksperiodeService.kt

```diff
@@ -162,16 +160,13 @@ class VedtaksperiodeService(
         val sisteTomPåVilkårSomAlltidSkalKunneBegrunnes =
             vilkårsvurdering.personResultater
                 .mapNotNull { personResultat ->
-
-                    val vilkårResultaterForAktørSomAlltidSkalKunneBegrunnes = personResultat.vilkårResultater.filter { listeAvVilkårSomAlltidSkalKunneBegrunnes.contains(it.vilkårType) && it.periodeFom != null }
-
-                    val vilkårResultaterForAktørMapSomAlltidSkalKunneBegrunnes = vilkårResultaterForAktørSomAlltidSkalKunneBegrunnes.groupByTo(mutableMapOf()) { it.vilkårType }.mapValues { it.value }
-
-                    vilkårResultaterForAktørMapSomAlltidSkalKunneBegrunnes
-                        .flatMap { (vilkårType, vilkårResultater) ->
-                            forskyvVilkårResultater(vilkårType, vilkårResultater).tilTidslinje().tilPerioderIkkeNull()
-                        }.mapNotNull { it.verdi.periodeTom }
-                        .maxOfOrNull { it }
+                    personResultat
+                        .forskyvVilkårResultater()
+                        .filterKeys { listeAvVilkårSomAlltidSkalKunneBegrunnes.contains(it) }
+                        .values
+                        .flatten()
+                        .mapNotNull { it.tom }
```

Comment:
Vi endret fra å finne seneste `periodeTom` i `VilkårResultat` til finne seneste `tom` i `Periode<VilkårResultat>`. For det er vel dato etter forskyvning vi er ute etter her og ikke original dato? @UyQuangNguyen